tests/runtime/abstract_atoms.m31

```diff
@@ -0,0 +1,4 @@
+rule A type;;
+let abstr = {z : A} z;;
+let a = match abstr with {y : Y} w :> judgement -> y end;;
+let abstr1 = abstract a a;;
```

Comment:
Maybe put in also a more direct test along the lines of `let x = fresh x : A in abstract x x`.